File: Matlab/utils/functions/bayer_bilinear.m
function Bay_demosaic = bayer_bilinear(mosaic)

conv_kernel_g = [1, 2, 1; 2, 4, 2; 1, 2, 1]/8;
conv_kernel_r = [1, 0, 1; 0, 4, 0; 1, 0, 1]/4;          
           
R_1 = conv2(mosaic(:,:,1), conv_kernel_r, "same");
R = conv2(R_1, conv_kernel_g, "same");
G = conv2(mosaic(:,:,2), conv_kernel_g, "same");
B_1 = conv2(mosaic(:,:,3), conv_kernel_r, "same");
B = conv2(B_1, conv_kernel_g, "same");

Bay_demosaic = cat(3,R,G,B);

File: Python/utils/functions/bayer_bilinear.py
import numpy as np
from utils.matlab_compat import conv2


def bayer_bilinear(mosaic):
    conv_kernel_g = np.array([[1, 2, 1], [2, 4, 2], [1, 2, 1]], dtype=np.float64) / 8.0
    conv_kernel_r = np.array([[1, 0, 1], [0, 4, 0], [1, 0, 1]], dtype=np.float64) / 4.0

    r_1 = conv2(mosaic[:, :, 0], conv_kernel_r, mode="same")
    r = conv2(r_1, conv_kernel_g, mode="same")
    g = conv2(mosaic[:, :, 1], conv_kernel_g, mode="same")
    b_1 = conv2(mosaic[:, :, 2], conv_kernel_r, mode="same")
    b = conv2(b_1, conv_kernel_g, mode="same")

    return np.stack([r, g, b], axis=2)
